<a href="https://colab.research.google.com/github/jatinsharma16/Capstone-Project-Unsupervised/blob/main/Netflix_movie_and_TV_show_clustering.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Project Name: Netflix movie and TV show clustering**

Project type: Unsupervised Machine Learning

Contribution: Individual

# **Project Summary:**

# **GITHUB Link**

link: https://github.com/jatinsharma16/Capstone-Project-Unsupervised

# **Problem Statement**

**PROBLEM OVERVIEW**

Netflix as a company is the biggest online platform present at time which provides us with the facility to enjoy movies or TV shows at any given time. In 2018, a report showed that the number of TV shows on Netflix has nearly tripled since 2010 whereas the number of movies has decreased by more than 2000 titles. This shows that how the consumers have shifted from consuming movies to TV shows on regular basis.

The main objective of this project is to bring out the insights hidden in the dataset and bringing out some answers to the questions like what type of content is preferred the most? Do people really prefer watching shows over movies? What are the type of content the people generally love to watch and many more. We will be performing Exploratory Data Analysis to answer these questions and finally we will be clustering the content based on the features which help the platform giving better recommendations and keep their customers engaged.

Giving better recommendations is really important for the OTT platforms since it will generate a interest in the customers and the more engaged the customer is the more revenue the platforms will generate through the customers resulting in profits for the company.

# **Lets Begin**

### Data Description

In [1]:
# Importing basic libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
# mounting the drive for some security reason so that no one can access the data of the file.
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [3]:
# Defining the path of the csv file and reading the content of the file
path = "/content/drive/MyDrive/Almabetter/Modules/Machine_Learning/Unsupervised_Capstone_project/NETFLIX MOVIES AND TV SHOWS CLUSTERING.csv"
df = pd.read_csv(path)

In [4]:
df.head()

show_id     type  title           director  \
0      s1  TV Show     3%                NaN   
1      s2    Movie   7:19  Jorge Michel Grau   
2      s3    Movie  23:59       Gilbert Chan   
3      s4    Movie      9        Shane Acker   
4      s5    Movie     21     Robert Luketic   

                                                cast        country  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...         Brazil   
1  Demián Bichir, Héctor Bonilla, Oscar Serrano, ...         Mexico   
2  Tedd Chan, Stella Chung, Henley Hii, Lawrence ...      Singapore   
3  Elijah Wood, John C. Reilly, Jennifer Connelly...  United States   
4  Jim Sturgess, Kevin Spacey, Kate Bosworth, Aar...  United States   

          date_added  release_year rating   duration  \
0    August 14, 2020          2020  TV-MA  4 Seasons   
1  December 23, 2016          2016  TV-MA     93 min   
2  December 20, 2018          2011      R     78 min   
3  November 16, 2017          2009  PG-13     80 min   
4    January 1, 2020          2008  PG-13    123 min   

                                           listed_in  \
0  International TV Shows, TV Dramas, TV Sci-Fi &...   
1                       Dramas, International Movies   
2                Horror Movies, International Movies   
3  Action & Adventure, Independent Movies, Sci-Fi...   
4                                             Dramas   

                                         description  
0  In a future where the elite inhabit an island ...  
1  After a devastating earthquake hits Mexico Cit...  
2  When an army recruit is found dead, his fellow...  
3  In a postapocalyptic world, rag-doll robots hi...  
4  A brilliant group of students become card-coun...

In [5]:
df.tail()

show_id     type                                    title     director  \
7782   s7783    Movie                                     Zozo  Josef Fares   
7783   s7784    Movie                                   Zubaan  Mozez Singh   
7784   s7785    Movie                        Zulu Man in Japan          NaN   
7785   s7786  TV Show                    Zumbo's Just Desserts          NaN   
7786   s7787    Movie  ZZ TOP: THAT LITTLE OL' BAND FROM TEXAS     Sam Dunn   

                                                   cast  \
7782  Imad Creidi, Antoinette Turk, Elias Gergi, Car...   
7783  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   
7784                                            Nasty C   
7785                         Adriano Zumbo, Rachel Khoo   
7786                                                NaN   

                                                country          date_added  \
7782  Sweden, Czech Republic, United Kingdom, Denmar...    October 19, 2020   
7783                                              India       March 2, 2019   
7784                                                NaN  September 25, 2020   
7785                                          Australia    October 31, 2020   
7786              United Kingdom, Canada, United States       March 1, 2020   

      release_year rating  duration  \
7782          2005  TV-MA    99 min   
7783          2015  TV-14   111 min   
7784          2019  TV-MA    44 min   
7785          2019  TV-PG  1 Season   
7786          2019  TV-MA    90 min   

                                              listed_in  \
7782                       Dramas, International Movies   
7783     Dramas, International Movies, Music & Musicals   
7784  Documentaries, International Movies, Music & M...   
7785                 International TV Shows, Reality TV   
7786                    Documentaries, Music & Musicals   

                                            description  
7782  When Lebanon's Civil War deprives Zozo of his ...  
7783  A scrappy but poor boy worms his way into a ty...  
7784  In this documentary, South African rapper Nast...  
7785  Dessert wizard Adriano Zumbo looks for the nex...  
7786  This documentary delves into the mystique behi...

In [6]:
df.shape

(7787, 12)

The dataset provided to us have 7787 rows which means we have the information about 7787 movies/shows present on the Netflix. Also 12 columns are present in the dataset which will provide us the important information needed for the analysis and clustering.

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7787 entries, 0 to 7786
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       7787 non-null   object
 1   type          7787 non-null   object
 2   title         7787 non-null   object
 3   director      5398 non-null   object
 4   cast          7069 non-null   object
 5   country       7280 non-null   object
 6   date_added    7777 non-null   object
 7   release_year  7787 non-null   int64 
 8   rating        7780 non-null   object
 9   duration      7787 non-null   object
 10  listed_in     7787 non-null   object
 11  description   7787 non-null   object
dtypes: int64(1), object(11)
memory usage: 730.2+ KB


In [8]:
df.describe()

release_year
count   7787.000000
mean    2013.932580
std        8.757395
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2018.000000
max     2021.000000

In [9]:
df.describe().shape

(8, 1)

In [10]:
# lets see all the columns present in the dataset
df.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')

<h1> Column Description

- **show_id**: Unique Id for every movie/TV show.
- **type**: tells whether the content is movie or a show.
- **title**: Title of the movie/show.
- **director**: Director of the content.
- **cast**: Cast or the actors involved in the movie/show.
- **country**: The country of production of the movie/show.
- **date_added**: The date when the movie/show was added to Netflix.
- **release_year**: Actual release year of the movie/show.
- **rating**: TV ratings of the show.
- **duration**: Total duration of the movie in minutes or in season for shows.
- **listed_in**: Genre of the movie/show.
- **desciption**: The summary description of the movie/show.

### Handling Duplicate Values

In [11]:
# Checking for duplicates
df[df.duplicated()]

Empty DataFrame
Columns: [show_id, type, title, director, cast, country, date_added, release_year, rating, duration, listed_in, description]
Index: []

Here we can see that no duplicate values are present in the dataset. Now we will check for the null values.

### Handling Null values

In [12]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2389
cast             718
country          507
date_added        10
release_year       0
rating             7
duration           0
listed_in          0
description        0
dtype: int64

## Exploratory Data Analysis